infrastructure/main.tf

```
@@ -1,5 +1,5 @@
 provider "azurerm" {
-  version = "=1.22.1"
+  version = "=1.41.0"
```

any reason you aren't using the latest version?
```suggestion
  version = "=1.42.0"
```